main/src/com/google/refine/exporters/CsvExporter.java

```diff
@@ -116,21 +113,53 @@ public void addRow(List<CellData> cells, boolean isHeader) {
                     String[] strings = new String[cells.size()];
                     for (int i = 0; i < strings.length; i++) {
                         CellData cellData = cells.get(i);
-                        strings[i] =
-                            (cellData != null && cellData.text != null) ?
-                            cellData.text :
-                            "";
+                        // If file is tsv and cell text contains internal tabs, then manually escape tabs
+                        if (separator.charAt(0) == '\t' && cellData != null && cellData.text != null && cellData.text.contains("\t")) {
+                            final String tabString = cellData.text.replace("\t", "\\t");
+                            strings[i] = tabString;
+                        }
+                        // If file is tsv & cell contains internal newlines, manually escape newlines
+                        else if (separator.charAt(0) == '\t' && cellData != null && cellData.text != null && cellData.text.contains("\n")) {
```

Comment:
Why is this second case specific to TSV files? I suspect the same reasoning could also apply to CSV, or with any other choice of separator, no?